src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingMiner.java

```diff
@@ -169,18 +175,38 @@ public int getDepthLimit(final Level level)
     {
         if (this.getBuildingLevel() == 1)
         {
-            return MAX_DEPTH_LEVEL_1;
+            return normalizeMaxDepth(MAX_DEPTH_LEVEL_1, level);
         }
         else if (this.getBuildingLevel() == 2)
         {
-            return MAX_DEPTH_LEVEL_2;
+            return normalizeMaxDepth(MAX_DEPTH_LEVEL_2, level);
         }
-        else if (this.getBuildingLevel() >= 3)
+        else if (this.getBuildingLevel() == 3)
+        {
+            return normalizeMaxDepth(MAX_DEPTH_MAX, level);
```

Comment:
Shouldn't this be using `MAX_DEPTH_LEVEL_3` so that they actually stop at 0 rather than going all the way to bedrock?